lib/iris/tests/unit/common/metadata/test_BaseMetadata.py

```diff
@@ -80,54 +146,1197 @@ def test__none_lhs_operand(self):
+
+    def test_negate_false(self):
+        return_value = False
+        with mock.patch.object(
+            self.cls, "__eq__", return_value=return_value
+        ) as mocker:
+            result = self.metadata.__ne__(self.other)
+
+        self.assertTrue(result)
+        (arg,), kwargs = mocker.call_args
+        self.assertEqual(self.other, arg)
+        self.assertEqual(dict(), kwargs)
+
+
+class Test__combine(tests.IrisTest):
+    def setUp(self):
+        self.kwargs = dict(
+            standard_name="standard_name",
+            long_name="long_name",
+            var_name="var_name",
+            units="units",
+            attributes=dict(one=sentinel.one, two=sentinel.two),
+        )
+        self.cls = BaseMetadata
+        self.metadata = self.cls(**self.kwargs)
+
+    def test_lenient(self):
+        return_value = sentinel._combine_lenient
+        other = sentinel.other
+        with mock.patch(
+            "iris.common.metadata._LENIENT", return_value=True
+        ) as mlenient:
+            with mock.patch.object(
+                self.cls, "_combine_lenient", return_value=return_value
+            ) as mcombine:
+                result = self.metadata._combine(other)
+
+                self.assertEqual(1, mlenient.call_count)
+                (arg,), kwargs = mlenient.call_args
+                self.assertEqual(self.metadata.combine, arg)
+                self.assertEqual(dict(), kwargs)
+
+                self.assertEqual(return_value, result)
+                self.assertEqual(1, mcombine.call_count)
+                (arg,), kwargs = mcombine.call_args
+                self.assertEqual(other, arg)
+                self.assertEqual(dict(), kwargs)
+
+    def test_strict(self):
+        dummy = sentinel.dummy
+        values = self.kwargs.copy()
+        values["standard_name"] = dummy
+        values["var_name"] = dummy
+        values["attributes"] = dummy
+        other = self.cls(**values)
+        with mock.patch("iris.common.metadata._LENIENT", return_value=False):
+            result = self.metadata._combine(other)
+
+        expected = [
+            None if values[field] == dummy else values[field]
+            for field in self.cls._fields
+        ]
+        self.assertEqual(expected, result)
+
+
+class Test__combine_lenient(tests.IrisTest):
+    def setUp(self):
+        self.cls = BaseMetadata
+        self.none = self.cls(*(None,) * len(self.cls._fields))._asdict()
+        self.names = dict(
+            standard_name=sentinel.standard_name,
+            long_name=sentinel.long_name,
+            var_name=sentinel.var_name,
+        )
+
+    def test_strict_units(self):
+        left = self.none.copy()
+        left["units"] = "K"
+        right = left.copy()
+        lmetadata = self.cls(**left)
+        rmetadata = self.cls(**right)
+
+        result = lmetadata._combine_lenient(rmetadata)
+        expected = list(left.values())
+        self.assertEqual(expected, result)
```

Comment:
As I was rolling through the testing (while took a looooong time) my testing approach slowly matured and I progressed.

The commutative testing angle came along mid-way through, and I've forgotten to apply it retrospectively - nice spot :eyes: 

I defo think it's worthwhile doing. So I'll sweep back through the tests and apply this where it might be applicable.

Reply:
Yeah, I think that I latched onto the commutative pattern of testing post `BaseMetadata` testing. So I'll add some similar coverage there now... thanks.